### PR TITLE
feature: Checkbox V3

### PR DESCRIPTION
**Pull request by olerichter00**
The type of this PR is: **Feature**

This PR resolves [CX-1701]

### Description

This PR adds the new V3 `Checkbox` and uses it depending on the current theme.

|Storybook 1|Storybook 2 |
|---|---|
| | |
|Simulator Screen Shot - iPhone 8 - 2021-08-17 at 12 00 30|Simulator Screen Shot - iPhone 8 - 2021-08-17 at 12 00 30|

### PR Checklist (tick all before merging)

- [x] I have included screenshots or videos to illustrate my changes, or I have not changed anything that impacts the UI.
- [x] I have tested my changes on **iOS** and **Android**.
- [x] I have added tests for my changes, or my changes don't require testing, or I have included a link to a separate Jira ticket covering the tests.
- [x] I have added a feature flag, or my changes don't require a feature flag. ([How do I add one?](https://github.com/artsy/eigen/blob/master/docs/developing_a_feature.md))
- [x] I have documented any follow-up work that this PR will require, or it does not require any.
- [x] I have added an app state migration, or my changes do not require one. ([What are migrations?](https://github.com/artsy/eigen/blob/master/docs/adding_state_migrations.md))
- [x] I have added a changelog entry below or my changes do not require one.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

- Add V3 Checkbox component (behind feature flag) - ole

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

-

</details>

[CX-1701]: https://artsyproduct.atlassian.net/browse/CX-1701

### CODE REVIEW

**Comment by ArtsyOpenSource** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  ❌ **An error occurred while validating your changelog, please make sure you provided a valid changelog.**
  </td>
    </tr>
  </tbody>
</table>



<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against d3b362352b49cb1234d9b0e56d0c3dcca5a408ae
</p>

File: src/palette/elements/Checkbox/Checkbox.tsx [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+        </CssTransition>
+
+        <Flex style={{ marginTop: 2 }}>
+          {!!text && <Text color={textColor}>{text}</Text>}
+          {!!subtitle && (
+            <Text variant="small" color={subtitleColor}>
+              {subtitle}
+            </Text>
+          )}
+          {children}
```
Comment (olerichter00) [inline src/palette/elements/Checkbox/Checkbox.tsx:99]: **Note:** Allowing children here for compatibility reasons to V2.

File: src/palette/elements/Checkbox/index.tsx [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+import React from "react"
+import { Checkbox as CheckboxV3, CheckboxProps } from "./Checkbox"
+import { Checkbox as CheckboxV2, CheckboxProps as CheckboxV2Props } from "./CheckboxV2"
+export * from "./Check"
+
+export const Checkbox: React.FC<CheckboxProps> = (props) => {
+  const allowV3 = usePaletteFlagStore((state) => state.allowV3)
+
+  if (allowV3) {
+    return <CheckboxV3 {...props} />
```
Comment (pvinis) [inline src/palette/elements/Checkbox/index.tsx:11]: could you wrap this in `ThemeV3`? I noticed that this is actually important for cases like this conditional.

**Review by pvinis — CHANGES_REQUESTED** [review]
Just one thing, then I think we are good to go!

File: src/palette/elements/Checkbox/index.tsx (hunk 2 above)
Comment (olerichter00) [reply src/palette/elements/Checkbox/index.tsx]: Done 👍

**Review by pvinis — APPROVED** [review]

**Comment by olerichter00** [comment]
#squashongreen

**Pull request merged**